.github/workflows/main.yml

```diff
@@ -206,22 +206,31 @@ jobs:
     runs-on: ubuntu-20.04
     needs: [ fablab-smoketest ]
     steps:
+      # allow time for investigation unless the workflow is cancelled or the smoketest succeeded or was skipped
       - name: Sleep If Failed
-        if: needs.fablab-smoketest.result != 'success' && needs.fablab-smoketest.result != 'skipped'
+        if: needs.fablab-smoketest.result != 'success' && needs.fablab-smoketest.result != 'skipped' && needs.fablab-smoketest.result != 'cancelled'
         run: |
           sleep 30m
 
+      # release cloud resources if the smoketest succeeded, failed, or was cancelled; unnecessary if skipped
       - name: Teardown Test Environment
-        if: always() && needs.fablab-smoketest.result != 'skipped'
```

Comment:
`always()` is workflow-level so this check is redundant to the job-level check at L205 and so never influences whether this step runs.